ProcessMaker/Jobs/SmartInbox.php

```diff
@@ -8,34 +8,33 @@
 use Illuminate\Foundation\Bus\Dispatchable;
 use Illuminate\Queue\InteractsWithQueue;
 use Illuminate\Queue\SerializesModels;
-use ProcessMaker\InboxRules\ApplyAction;
-use ProcessMaker\InboxRules\MatchingTasks;
-use ProcessMaker\Models\InboxRule;
+use Facades\ProcessMaker\InboxRules\ApplyAction;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 use ProcessMaker\Models\ProcessRequestToken;
 
 class SmartInbox implements ShouldQueue
 {
     use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
 
-    public $task;
+    public $incomingTaskId;
     /**
      * Create a new job instance.
      */
-    public function __construct(ProcessRequestToken $task)
+    public function __construct(int $incomingTaskId)
     {
-        $this->task = $task;
+        $this->incomingTaskId = $incomingTaskId;
     }
 
     /**
      * Execute the job.
      */
     public function handle(): void
-    {
-        $inboxRule = new InboxRule();
-        $matchingTasks = new MatchingTasks($inboxRule);
-        $matchResult = $matchingTasks->check($this->task);
-        if($matchResult){
-            //Here calls ApplyAction Class
+    {   
+        try {
+            $incomingTask = ProcessRequestToken::findOrFail($this->incomingTaskId);
+            ApplyAction::applyActionOnTask($incomingTask);
+        } catch (ModelNotFoundException $e) {
```

Comment:
You don't need try/catch here. Let the global error handler take care of it

Reply:
Solved.